client/logmon/plugin.go

```diff
@@ -21,18 +21,25 @@ func LaunchLogMon(logger hclog.Logger, reattachConfig *plugin.ReattachConfig) (L
 		return nil, nil, err
 	}
 
-	client := plugin.NewClient(&plugin.ClientConfig{
+	conf := &plugin.ClientConfig{
 		HandshakeConfig: base.Handshake,
-		Reattach:        reattachConfig,
 		Plugins: map[string]plugin.Plugin{
 			"logmon": &Plugin{},
 		},
-		Cmd: exec.Command(bin, "logmon"),
 		AllowedProtocols: []plugin.Protocol{
 			plugin.ProtocolGRPC,
 		},
 		Logger: logger,
-	})
+	}
+
+	// Only set one of Cmd or Reattach
+	if reattachConfig == nil {
```

Comment:
❓ Doesn't this mean that if a logmon process has been `pkill`'d or crashed, we won't respawn it?

Reply:
05bae8d took the cowards way out: return a recoverable error on logmon launch failures and rely upon the task restart machinery to launch a new logmon.

Trying to start a new logmon in place will take more code and testing, and I didn't want to hold up beta3 for it.